### PR TITLE
feat(cloud-masthead): masthead, adjust the position of the CTA buttons based on the new Mobile specs

### PR DESCRIPTION
**Pull request by huntermacd**
### Related Ticket(s)

https://jsw.ibm.com/browse/HC-1829

### Description

This diff arranges the Cloud profile menu items appropriately to match the design at both narrow and wide viewport widths and styles them.

### Changelog

**New**

- cloud-left-nav-item.ts - extended component in order to apply new styles

**Changed**

- cloud-masthead-composite.ts - adjust markup to render profile items in correct location
- cloud-masthead.scss - new styles added

**Removed**

- n/a

### CODE REVIEW

**Review by proeung — APPROVED** [review]
Looks good overall!

The CTA buttons are now moved to the hamburger menu dropdown on mobile breakpoints and the "User" dropdown button is visible when you're authenticated.

**Anonymous State**
Screen Shot 2021-04-21 at 4 20 31 PM

Authenticated State
Screen Shot 2021-04-21 at 4 20 49 PM

Screen Shot 2021-04-21 at 4 21 00 PM

I think we'll just need an additional code review and have Lila take a look at this PR as well.

**Comment by ltitle** [comment]
@huntermacd @proeung CTA behavior looks good to me. Thanks!

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

File: packages/web-components/src/components/masthead/cloud/cloud-masthead.scss [hunk 1]
```diff
@@ -110,3 +116,17 @@
...
+:host(#{$dds-prefix}-cloud-left-nav-item) {
+  @extend :host(#{$dds-prefix}-left-nav-item);
+
+  &.left-nav-cta .bx--side-nav__link .bx--side-nav__link-text {
+    color: $link-01;
+  }
+}
+
+:host(#{$dds-prefix}-cloud-masthead-profile) .bx--header__menu-title[aria-expanded='true'] + .bx--header__menu {
+  @include carbon--breakpoint-down(320px) {
```
Comment (annawen1) [inline packages/web-components/src/components/masthead/cloud/cloud-masthead.scss:129]: Wondering for the profile menu - do we only want the menu to be full width when on 320px or smaller? I only see the menu going full width on the iphone 5/SE currently. Maybe it makes more sense to do:

``` 
@include carbon--breakpoint(sm) {
    width: 100vw;
  }
```

Comment (huntermacd) [reply packages/web-components/src/components/masthead/cloud/cloud-masthead.scss]: Indeed. Not sure from the design docs at exactly which breakpoint the menu moves to full-width. I combined our ideas. :) https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5856/commits/a5050a3e1387c45f7a8662a2adafe9eb70a4fcb8 Thanks!

**Comment by shixiedesign** [comment]
Thanks @proeung for walking me through! All good to merge

**Pull request merged**